fenix/app/src/main/java/org/mozilla/fenix/shopping/store/ReviewQualityCheckState.kt

```diff
@@ -159,6 +159,17 @@ sealed interface ReviewQualityCheckState : State {
                     NEEDS_ANALYSIS, REANALYZING, UP_TO_DATE
                 }
             }
+
+            /**
+             * Denotes a product is not available.
+             *
+             * @property reported Whether or not current user has reported the product is back in stock.
+             * @property anotherUserReported Whether or not another user has reported the product is back in stock.
+             */
+            data class NotAvailable(
```

Comment:
This should be a child of the Error sealed interface, similar to `NotEnoughReviews` and `UnsupportedProductTypeError`

Reply:
Also, Looking at the [designs](https://www.figma.com/file/VE5xayz3yOSB5qOATUU9Im/Mobile-Shopping-Experience-MVP?type=design&node-id=5384-23225&mode=design&t=l1duuWVGcjQZAxvB-0) we want to represent 3 different states, 

`ProductNotAvailable` -> When `deleted_product` is returned true by the analysis.  Shows the button also. 
`ThanksForReporting` -> When button is clicked on `ProductNotAvailable`
`ProductAlreadyReported` -> When `deleted_product_reported` is returned true by the analysis api.

With this, they become objects in the Error sealed interface, similar to other Error states and the UI is 1:1 mapped.

Reply:
Updated the 3 states to be Error states